### PR TITLE
Blog app - Add forms

### PR DESCRIPTION
**Pull request by amejid**
## Milestone 7: Add forms

In this PR, the following were implemented:-

- functionality to create a new post
- functionality to give likes and comments on a post

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -17,4 +33,8 @@ def set_post
   def set_user
     @user = User.find(params[:user_id])
   end
+
+  def post_params
+    params.require(:post).permit(:title, :text)
+  end
```
Comment (rloterh) [inline app/controllers/posts_controller.rb:39]: - [ ] OPTIONAL
It's a good rule to mark strong params method, `post_params`, as _private_. Kindly endeavor to improve upon the same.

File: app/controllers/comments_controller.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    @user = User.find(params[:user_id])
+  end
+
+  def set_post
+    @post = Post.find(params[:post_id])
+  end
+
+  def comment_params
+    params.require(:comment).permit(:text)
+  end
```
Comment (rloterh) [inline app/controllers/comments_controller.rb:32]: - [ ] OPTIONAL
It's a good rule to mark strong params method, `comment_params`, as _private_. Kindly endeavor to improve upon the same.

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Abdelmejid,  

Your project is complete, great job for nailing it right🥇 There is nothing else to say other than... it's time to merge it 🍾🍾 Congratulations!  🎉💯🌟 

## Highlights 👍
- Form Controllers actions and Views properly implemented

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**You can also consider:**
- It's recommended that you add an MIT license, or any other similar license to your ReadMe.
Implementing the above change(s) will make your ReadMe/ repository stand out even better 😉

Keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (amejid) [reply app/controllers/posts_controller.rb:39]: Thanks for the review @rloterh , but the post_params is under a private keyword.

File: app/controllers/comments_controller.rb (hunk 2 above)
Comment (amejid) [reply app/controllers/comments_controller.rb:32]: The same thing here.

**Comment by amejid** [comment]
Thanks for reviewing and approving my work, I'll implement the changes accordingly.

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (rloterh) [reply app/controllers/posts_controller.rb:39]: @amejid Thanks for clarifying the same. Cheers, and happy coding! 😃

**Pull request merged**